### ISSUE TITLE
E1024 Fn::Cidr should allow Fn::If that returns valid types

### ISSUE THREAD

**Issue by elliotsegler** [aws-cloudformation/cfn-lint#2139]
*cfn-lint version: (`0.52.0+`)*

*Description of issue.*

We cut down Subnets from the VPC CIDR by scoping down CIDRs in increasing sizes and using the `Fn::Cidr` function with select to pick the appropriate subnet.

Currently, it looks like `E1024` allows a Cidr Range, Ref, GetAtt, Sub or a Select function, but doesn't allow an If function that returns a valid type. The below syntax is valid in CloudFormation, and because both possible return values are appropriately typed should not return an error.

```yaml
  ExampleSubnet:
    Type: AWS::EC2::Subnet
    Properties:
      # /28 block #1 (VPC Allocation), inside /26 block #3, inside /24 block #3 (prod) or block #15 (non-prod)
      CidrBlock: !Select [ 1, !Cidr [
        !Select [ 3, !Cidr [
          !If [
            IsProd,
            !Select [ 3, !Cidr [ !GetAtt MyVPC.CidrBlock, 4, 8 ]],
            !Select [ 15, !Cidr [ !GetAtt MyVPC.CidrBlock, 16, 8 ]]
          ],
          4, 6
        ]],
        2, 4
      ]]
      VpcId: !Ref MyVPC
```

Tested on cfn-lint versions:
* `0.52.0`
* `0.54.2`
